k8s/externalsnapshotter/externalsnapshotter.go

```diff
@@ -0,0 +1,147 @@
+package externalsnapshotter
+
+import (
+	"fmt"
+	"os"
+	"sync"
+
+	"github.com/kubernetes-csi/external-snapshotter/v2/pkg/client/clientset/versioned/typed/volumesnapshot/v1beta1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+var (
+	instance Ops
+	once     sync.Once
+)
+
+// Ops is an interface to perform operations on external storage resources.
+type Ops interface {
+	SnapshotOps
+
+	// SetConfig sets the config and resets the client
+	SetConfig(config *rest.Config)
+}
+
+// Instance returns a singleton instance of the client.
+func Instance() Ops {
+	once.Do(func() {
+		if instance == nil {
+			instance = &Client{}
+		}
+	})
+	return instance
+}
+
+// SetInstance replaces the instance with the provided one. Should be used only for testing purposes.
+func SetInstance(i Ops) {
+	instance = i
+}
+
+// New builds a new client for the given config.
+func New(client rest.Interface) *Client {
+	return &Client{
+		snap: client,
+	}
+}
+
+// NewForConfig builds a new client for the given config.
+func NewForConfig(c *rest.Config) (*Client, error) {
+	client, err := v1beta1.NewForConfig(c)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Client{
+		client: client,
+	}, nil
+}
+
+// NewInstanceFromConfigFile returns new instance of client by using given
+// config file
+func NewInstanceFromConfigFile(config string) (Ops, error) {
+	newInstance := &Client{}
+	err := newInstance.loadClientFromKubeconfig(config)
+	if err != nil {
+		return nil, err
+	}
+	return newInstance, nil
+}
+
+// Client is a wrapper for the external-storage client.
+type Client struct {
+	config *rest.Config
+	snap   rest.Interface
+
+	client v1beta1.SnapshotV1beta1Interface
+}
+
+// SetConfig sets the config and resets the client
+func (c *Client) SetConfig(cfg *rest.Config) {
+	c.config = cfg
+	c.snap = nil
+}
+
+// initClient the k8s client if uninitialized
+func (c *Client) initClient() error {
+	if c.snap != nil {
```

Comment:
Do we even need `snap` variable? We should just be checking `if c.client != nil` and remove `snap` from Client.

Reply:
Yes, there is no need for snap. Updated.